genesis/engine/couplers/sap_coupler.py

```diff
@@ -1489,6 +1542,233 @@ def active_solvers(self):
+    ):
+        self.n_constraints[None] = 0
+        self.Jt.fill(0.0)
+        # Now all batches have the same constraints
+        # TODO maybe support different constraints for each batch in the future
+        dt2 = self.sim._substep_dt**2
+        for i_b, i_e in ti.ndrange(self._B, self.rigid_solver.n_equalities):
+            if equalities_info.eq_type[i_e, i_b] == gs.EQUALITY_TYPE.JOINT:
+                i_c = ti.atomic_add(self.n_constraints[None], 1)
+                self.constraints[i_c].batch_idx = i_b
+                I_joint1 = (
+                    [equalities_info.eq_obj1id[i_e, i_b], i_b]
+                    if ti.static(static_rigid_sim_config.batch_joints_info)
+                    else equalities_info.eq_obj1id[i_e, i_b]
+                )
+                I_joint2 = (
+                    [equalities_info.eq_obj2id[i_e, i_b], i_b]
+                    if ti.static(static_rigid_sim_config.batch_joints_info)
+                    else equalities_info.eq_obj2id[i_e, i_b]
+                )
+                i_dof1 = joints_info.dof_start[I_joint1]
+                i_dof2 = joints_info.dof_start[I_joint2]
+                self.constraints[i_c].i_dof1 = i_dof1
+                self.constraints[i_c].i_dof2 = i_dof2
+                self.constraints[i_c].sap_info.k = self.stiffness
+                self.constraints[i_c].sap_info.R_inv = dt2 * self.stiffness
+                self.constraints[i_c].sap_info.R = 1.0 / self.constraints[i_c].sap_info.R_inv
+                self.constraints[i_c].sap_info.v_hat = 0.0
+                self.Jt[i_c, i_dof1] = 1.0
+                self.Jt[i_c, i_dof2] = -1.0
+
+    @ti.func
+    def compute_regularization(self):
+        dt_inv = 1.0 / self.sim._substep_dt
+        q = ti.static(self.rigid_solver.dofs_state.pos)
+        sap_info = ti.static(self.constraints.sap_info)
+        for i_c in range(self.n_constraints[None]):
+            i_b = self.constraints[i_c].batch_idx
+            g0 = q[self.constraints[i_c].i_dof1, i_b] - q[self.constraints[i_c].i_dof2, i_b]
+            self.constraints[i_c].sap_info.v_hat = -g0 * dt_inv
+            W = self.compute_delassus(i_c)
+            self.compute_constraint_regularization(sap_info, i_c, W, self.sim._substep_dt)
+
+    @ti.func
+    def compute_delassus_world_frame(self):
+        self.coupler.rigid_solve_jacobian(
+            self.Jt, self.M_inv_Jt, self.n_constraints[None], self.constraints.batch_idx, 1
+        )
+        self.W.fill(0.0)
+        for i_c, i_d in ti.ndrange(self.n_constraints[None], self.rigid_solver.n_dofs):
+            self.W[i_c] += self.M_inv_Jt[i_c, i_d] * self.Jt[i_c, i_d]
+
+    @ti.func
+    def compute_delassus(self, i_c):
+        return self.W[i_c]
+
+    @ti.func
+    def compute_Jx(self, i_c, x):
+        i_b = self.constraints[i_c].batch_idx
+        i_dof1 = self.constraints[i_c].i_dof1
+        i_dof2 = self.constraints[i_c].i_dof2
+        Jx = x[i_b, i_dof1] - x[i_b, i_dof2]
+        return Jx
```

Comment:
Just return directly. This temporary does not bring any value because the name of this function is on point.